### PR TITLE
Externalize leader-elector config for multi-provisioner deployments

### PR DESCRIPTION
**Pull request by npramod**
* Facilitates tuning deployement to different cloud provisioning api SLAs
For using just the controller as a library to build other provisioners, externalizing leader election config from controller would help one in tuning leader election to suit different cloud/IaaS api response times. Default/hardcoded values of 15s do not generalize well.

### CODE REVIEW

File: demo/hostpath-provisioner/hostpath-provisioner.go [hunk 1]
```diff
@@ -121,6 +126,6 @@ func main() {
 
 	// Start the provision controller which will dynamically provision hostPath
 	// PVs
-	pc := controller.NewProvisionController(clientset, resyncPeriod, provisionerName, hostPathProvisioner, serverVersion.GitVersion, exponentialBackOffOnError, failedRetryThreshold)
+	pc := controller.NewProvisionController(clientset, resyncPeriod, provisionerName, hostPathProvisioner, serverVersion.GitVersion, exponentialBackOffOnError, failedRetryThreshold, leasePeriod, renewDeadline, retryPeriod, termLimit)
```
Comment (wongma7) [inline demo/hostpath-provisioner/hostpath-provisioner.go]: you  may revert the changes in this file, I am still figuring out how to keep this up-to-date. 

otherwise lgtm.

**Comment by wongma7** [comment]
thanks. Please squash into one commit and I will merge

BTW in the near future we are going to move to a new repo so you will want to update your import paths. It's to more cleanly separate the library+the nfs portion. 

Probably the move will be announced on sig-storage mialing list

**Comment by npramod** [comment]
Yes, I am aware of the move. Will keep an eye out for the announcement. Thanks for the quick response.

**Comment by npramod** [comment]
@wongma7 Is the CI failure fine? Looks like hostpath demo is failing

**Comment by wongma7** [comment]
Yeah, it's okay. I want to exclude the demo from the CI since it has its own dependencies and such. thanks
